Original xml.m4 is useless for multiarch systems like Solaris.
This patch tries to fix that.

--- clamav-0.100.1/m4/reorganization/libs/xml.m4.~1~	2018-06-27 22:12:10.000000000 +0000
+++ clamav-0.100.1/m4/reorganization/libs/xml.m4	2018-07-26 13:57:00.326537652 +0000
@@ -5,6 +5,7 @@
 want_xml=$enableval, want_xml="auto")
 
 XML_HOME=""
+xmlconfig=""
 if test "X$want_xml" != "Xno"; then
   AC_MSG_CHECKING([for libxml2 installation])
   AC_ARG_WITH([xml],
@@ -14,6 +15,10 @@
   )
 fi

+AC_PATH_PROG([xmlconfig], [xml2-config])
+
+if test "x$xmlconfig" = x ; then
+
 AS_IF([test "x$with_xml_val" = "xno"], [XML_HOME=""],
   [test "x$with_xml_val" = "xyes"], [XML_HOME="/usr/local"],
   [XML_HOME="$with_xml_val"])
@@ -31,21 +36,25 @@
 else
   AC_MSG_RESULT([not found])
 fi
+xmlconfig=$XML_HOME/bin/xml2-config
+
+fi
 
 found_xml="no"
 XMLCONF_VERSION=""
 XML_CPPFLAGS=""
 XML_LIBS=""
-if test "x$XML_HOME" != "x"; then
+if test "x$xmlconfig" != "x"; then
   AC_MSG_CHECKING([xml2-config version])
-  XMLCONF_VERSION="`$XML_HOME/bin/xml2-config --version`"
+  XMLCONF_VERSION="`$xmlconfig --version`"
   if test "x%XMLCONF_VERSION" != "x"; then
     AC_MSG_RESULT([$XMLCONF_VERSION])
     found_xml="yes"
-    XML_CPPFLAGS="`$XML_HOME/bin/xml2-config --cflags`"
-    XML_LIBS="`$XML_HOME/bin/xml2-config --libs`"
+    XML_CPPFLAGS="`$xmlconfig --cflags`"
+    XML_LIBS="`$xmlconfig --libs`"
     AS_ECHO("$XML_CPPFLAGS")
     AS_ECHO("$XML_LIBS")
+    XML_HOME=`$xmlconfig --prefix`
   else
     AC_MSG_ERROR([xml2-config failed])
   fi
